test/integration_test/migration_test.go

```diff
@@ -52,41 +59,76 @@ func BasicMigration(tc *types.TestCase) func(*testing.T) {
 			t.Skip()
 		}
 
-		logrus.Infof("Creating portworx daemonset spec")
 		ds := testutil.GetExpectedDaemonSet(t, "migration/simple-daemonset.yaml")
 		require.NotNil(t, ds)
 
-		_, err := appops.Instance().CreateDaemonSet(ds, metav1.CreateOptions{})
-		require.NoError(t, err)
+		nodeNameWithLabel := ci_utils.AddLabelToRandomNode(t, labelKeyPXEnabled, ci_utils.LabelValueFalse)
+
+		testMigration(t, ds)
+
+		ci_utils.RemoveLabelFromNode(t, nodeNameWithLabel, labelKeyPXEnabled)
+	}
+}
+
+func MigrationWithoutNodeAffinity(tc *types.TestCase) func(*testing.T) {
+	return func(t *testing.T) {
+		if tc.ShouldSkip() {
+			t.Skip()
+		}
+
+		ds := testutil.GetExpectedDaemonSet(t, "migration/simple-daemonset.yaml")
+		require.NotNil(t, ds)
+
+		ds.Spec.Template.Spec.Affinity = nil
+
+		testMigration(t, ds)
+	}
+}
+
+func testMigration(t *testing.T, ds *appsv1.DaemonSet) {
+	logrus.Infof("Creating portworx daemonset spec")
+	_, err := appops.Instance().CreateDaemonSet(ds, metav1.CreateOptions{})
+	require.NoError(t, err)
 
-		err = appops.Instance().ValidateDaemonSet(ds.Name, ds.Namespace, ci_utils.DefaultValidateDeployTimeout)
-		require.NoError(t, err)
+	err = appops.Instance().ValidateDaemonSet(ds.Name, ds.Namespace, ci_utils.DefaultValidateDeployTimeout)
+	require.NoError(t, err)
 
-		logrus.Infof("Restarting portworx operator to trigger migration")
-		restartPortworxOperator(t)
+	logrus.Infof("Restarting portworx operator to trigger migration")
+	restartPortworxOperator(t)
 
-		pxImageMap, err := getPortworxImagesFromDaemonSet(t, ds)
-		require.NoError(t, err)
+	pxImageMap, err := getPortworxImagesFromDaemonSet(t, ds)
```

Comment:
nit: could create a new `daemonset.go` in utils and move this function there? similar to other utils functions that could be shared.

Reply:
Actually I changed the logic of this. Instead of having a hard-coded version in the daemonset spec, I am rather creating the DS with the version URL we have passed to the test suite. So relying on PxSpecImages to fill the px version instead of the daemonset version to construct the pxSpecImages map.